### ISSUE TITLE
Inherited FK constraint prevents rollback

### ISSUE THREAD

**Issue by yhuelf** [dalibo/emaj#44]
Maybe this should be added into the "usage limit" section of the documentation.

```
postgres@emaj=# create table foo(id int generated always as identity, p int primary key);
CREATE TABLE
postgres@emaj=# create table bar(id int generated always as identity,
foo_id int references foo(p),
insert_time timestamptz default now(),
primary key(id, insert_time)
) partition by range(insert_time);
CREATE TABLE

postgres@emaj=# create table bar_2021 partition of bar for values from ('2021-01-01') to ('2021-12-31');
CREATE TABLE

postgres@emaj=# insert into foo(p) values(5);
INSERT 0 1
postgres@emaj=# insert into bar(foo_id) values(5);
INSERT 0 1

postgres@emaj=# SELECT emaj.emaj_create_group('toto', true);
 emaj_create_group 
-------------------
                 1
(1 row)

postgres@emaj=# SELECT emaj.emaj_assign_table('public', 'bar_2021', 'toto');
NOTICE:  table "bar_2021_log" does not exist, skipping
NOTICE:  trigger "emaj_log_trg" for relation "public.bar_2021" does not exist, skipping
NOTICE:  trigger "emaj_trunc_trg" for relation "public.bar_2021" does not exist, skipping
 emaj_assign_table 
-------------------
                 1
(1 row)

postgres@emaj=# select * from bar;
 id | foo_id |          insert_time          
----+--------+-------------------------------
  1 |      5   |  2021-12-23 08:06:49.041451+01
(1 row)

postgres@emaj=# SELECT emaj.emaj_start_group('toto', 'M1');
WARNING:  _check_fk_groups: The foreign key "bar_foo_id_fkey" on the table "public.bar_2021" references the table "public.foo" that is outside the groups (toto).
 emaj_start_group 
------------------
                1
postgres@emaj=# insert into foo(p) values(7);
INSERT 0 1
postgres@emaj=# insert into bar(foo_id) values(7);
INSERT 0 1
postgres@emaj=# select * from bar;
 id | foo_id |          insert_time          
----+--------+-------------------------------
  1 |      5   |   2021-12-23 08:06:49.041451+01
  2 |      7   |   2021-12-23 08:10:22.538028+01
(2 rows)

postgres@emaj=# SELECT emaj.emaj_rollback_group('toto','M1');
ERROR:  cannot drop inherited constraint "bar_foo_id_fkey" of relation "bar_2021"
CONTEXT:  SQL statement "ALTER TABLE public.bar_2021 DROP CONSTRAINT bar_foo_id_fkey"
PL/pgSQL function emaj._handle_trigger_fk_tbl(text,text,text,text) line 48 at EXECUTE
SQL statement "SELECT emaj._handle_trigger_fk_tbl('DROP_FK', v_fullTableName, r_step.rlbp_object)"
PL/pgSQL function emaj._rlbk_session_exec(integer,integer) line 67 at PERFORM
SQL statement "SELECT emaj._rlbk_session_exec(v_rlbkId, 1)"
PL/pgSQL function emaj._rlbk_groups(text[],text,boolean,boolean,boolean) line 32 at PERFORM
SQL statement "SELECT *
        FROM emaj._rlbk_groups(array[p_groupName], p_mark, FALSE, FALSE, coalesce(p_isAlterGroupAllowed, FALSE))"
PL/pgSQL function emaj.emaj_rollback_group(text,text,boolean) line 7 at RETURN QUERY
```

**Comment by beaud76** [comment]
Thank You for the report.

Right now, it looks like a bug rather than a limitation. I will study this case.

**Comment by yhuelf** [comment]
A workaround is to set the FK constraint for the partitions instead of the parent table.

**Comment by beaud76** [comment]
Fixed by commits 74bf90a5 and 37721227.
